src/test/java/core/basesyntax/operation/impl/PurchaseOperationHandlerTest.java

```diff
@@ -0,0 +1,100 @@
+package core.basesyntax.operation.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImp;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.operation.OperationHandler;
+import core.basesyntax.service.FruitService;
+import core.basesyntax.service.impl.FruitServiceImpl;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class PurchaseOperationHandlerTest {
+    private FruitDao fruitDao;
+    private FruitService fruitService;
+    private OperationHandler operationHandler;
+
+    @Before
+    public void setUp() {
+        fruitDao = new FruitDaoImp();
+        fruitService = new FruitServiceImpl(fruitDao);
+        operationHandler = new PurchaseOperationHandler(fruitDao, fruitService);
+    }
+
+    @Test
+    public void handle_updateExistedFruit_ok() {
+        FruitTransaction existedInStorage = new FruitTransaction();
+        existedInStorage.setFruit("banana");
+        existedInStorage.setOperation(FruitTransaction.Operation.BALANCE);
+        existedInStorage.setQuantity(100);
+        Storage.warehouse.add(existedInStorage);
+        FruitTransaction update = new FruitTransaction();
+        update.setFruit("banana");
+        update.setOperation(FruitTransaction.Operation.PURCHASE);
+        update.setQuantity(50);
+        operationHandler.handle(update);
+        FruitTransaction expected = new FruitTransaction();
+        expected.setFruit("banana");
+        expected.setOperation(FruitTransaction.Operation.BALANCE);
+        expected.setQuantity(50);
+        FruitTransaction actual = fruitDao.get("banana");
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void handle_parameterNull_notOk() {
+        assertThrows(RuntimeException.class, () -> {
+            operationHandler.handle(null);
+        });
+    }
+
+    @Test
+    public void handle_parameterNotSetFields_notOk() {
+        assertThrows(RuntimeException.class, () -> {
+            operationHandler.handle(new FruitTransaction());
+        });
+    }
+
+    @Test
+    public void handle_quantityNull_notOk() {
+        FruitTransaction fruitTransaction = new FruitTransaction();
+        fruitTransaction.setQuantity(0);
```

Comment:
```suggestion
        fruitTransaction.setQuantity(null);
```

null or zero?

Reply:
I can't set null for variable int.

Reply:
then rename method
<img width="865" alt="image" src="https://user-images.githubusercontent.com/68428976/173824910-2e97652e-0c3c-4b11-b67e-6d1526ed8724.png">

Reply:
Than rename method 
<img width="881" alt="image" src="https://user-images.githubusercontent.com/68428976/173825110-37d5db1e-3bf6-4c7f-82f1-0da190f8890a.png">